### PR TITLE
Add --allow-extra-keys to check-source-has-meta-keys

### PR DESCRIPTION
**Pull request by BAntonellini**

### CODE REVIEW

File: dbt_checkpoint/check_exposure_has_meta_keys.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    add_default_args,
+    add_meta_keys_args,
+    get_dbt_manifest,
+    get_exposures,
+    red,
+    yellow,
+)
+
+
+def has_meta_key(
```
Comment (JFrackson) [inline dbt_checkpoint/check_exposure_has_meta_keys.py]: This function seems pretty similar to what we need to do with models, sources, and exposures. Did you think about generalizing and then adding this functionality as a util?

File: dbt_checkpoint/check_exposure_has_meta_keys.py
Comment (alfredodimassimo) [inline dbt_checkpoint/check_exposure_has_meta_keys.py:1]: Looks great overall, but would you be able to add tests to this one, looks like it's coming back with 0% coverage

File: dbt_checkpoint/check_exposure_has_meta_keys.py (hunk 1 above)
Comment (BAntonellini) [reply dbt_checkpoint/check_exposure_has_meta_keys.py]: Hey @JFrackson, sources and exposures ones are in part similar, though sources contain a schema+table meta union.

Models`has_meta_key` is almost entirely different IMO

File: dbt_checkpoint/check_exposure_has_meta_keys.py
Comment (BAntonellini) [reply dbt_checkpoint/check_exposure_has_meta_keys.py:1]: @alfredodimassimo `test_check_exposure_has_meta_keys` was part of #181

**Review by alfredodimassimo — APPROVED** [review]
LGTM!

**Pull request merged**